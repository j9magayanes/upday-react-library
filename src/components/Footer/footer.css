
footer {
    height: 80px;
    font-size: 0.625rem;
    line-height: 0.8125rem;
    text-align: center;
    color: #FFFFFF;
    background: #50D4F5;
    margin-top: 23px;
    display: flex;
    flex-direction: column;
    justify-content: center;
}

footer ul {
    font-weight: bold;
    display: flex;
    margin-right: auto;
    margin-left: auto;
    list-style: none;
    
}

footer .padded-lr {
    padding-left: 20px;
    padding-right: 20px;
    padding-bottom: 4px;
}

footer a {
    padding: 1em;
    color: #FFFFFF
}

footer li {
    font-family: Roboto, sans-serif;
    cursor: pointer;
    margin-right: pxToRem(32);
}
footer p {
    font-family: Roboto, sans-serif;
    padding: 4px; 
}
