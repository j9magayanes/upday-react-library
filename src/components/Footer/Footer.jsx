import React from 'react';
import './footer.css';
import 'upday-web-components/components/listComponent';
/* interface propType {
  footerItemName: String,
  footerItemCallback: Function
} */

export const Footer = ({ args }) => {
  return (
    <div className='padded-lr'>
     
      <list-component items={args.items} />
    </div>
  );
};
