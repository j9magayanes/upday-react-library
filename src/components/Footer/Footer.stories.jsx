import React from 'react';

import { Footer } from './Footer';


export default {
  title: 'Form/Footer',
  component: Footer,
};

const Template = () =><Footer args= {{
  items: '[{"name": "Privacy Policy", "href": "http://localhost:3000/privacy_policy#26fc5702-5d0a-4b14-8135-4bc49ff2fa49"}, {"name": "Privacy Settings", "callback": ""}, {"name": "Imprint", "href": "http://localhost:3000/imprint"}]',
}}/>;


export const Default = Template.bind({
});