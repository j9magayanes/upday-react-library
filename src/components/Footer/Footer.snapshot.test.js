import React from 'react';
import renderer from 'react-test-renderer';
import { Footer } from './Footer';

test('renders Footer component correctly', () => {
  const props = {
    args: {
      items: ['Item 1', 'Item 2', 'Item 3'],
    },
  };

  const component = renderer.create(<Footer {...props} />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
