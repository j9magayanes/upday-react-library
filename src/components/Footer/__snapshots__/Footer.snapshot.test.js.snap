// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Footer component correctly 1`] = `
<div
  className="padded-lr"
>
  <list-component
    items={
      Array [
        "Item 1",
        "Item 2",
        "Item 3",
      ]
    }
  />
</div>
`;
